rinja_parser/src/node.rs

```diff
@@ -90,10 +90,7 @@ impl<'a> Node<'a> {
             }
         };
 
-        let (i, _) = s.nest(j)?;
-        let result = func(i, s);
-        s.leave();
-        let (i, node) = result?;
```

Comment:
It is much clearer what is going on with you change, and the code does not much more `nom`-like. Clear advancement!